app/src/main/java/com/infomaniak/mail/data/models/Folder.kt

```diff
@@ -0,0 +1,134 @@
+
+    @SerializedName("unread_count")
+    var unreadCount: Int = 0
+
+    @SerializedName("total_count")
+    var totalCount: Int = 0
+
+    @SerializedName("is_fake")
+    var isFake: Boolean = false
+
+    @SerializedName("is_collapsed")
+    var isCollapsed: Boolean = false
+
+    @SerializedName("is_favorite")
+    var isFavorite: Boolean = false
+    var separator: String = ""
+    var children: RealmList<Folder> = realmListOf()
+
+    /**
+     * Local
+     */
+    var threads: RealmList<Thread> = realmListOf()
+    var parentLink: Folder? = null // TODO
+
+    fun updateAndSelect(mailboxUuid: String) {
+        fetchThreadsFromApi(mailboxUuid)
+        select()
+    }
+
+    fun select() {
+        MailRealm.mutableCurrentFolderIdFlow.value = id
+    }
+
+    private fun fetchThreadsFromApi(mailboxUuid: String) {
+        // Get current data
+        Log.d("API", "Threads: Get current data")
+        val threadsFromRealm = threads
+        // TODO: Handle connectivity issues. If there is no Internet, this list will be empty, so all Realm Threads will be deleted. We don't want that.
+        val threadsFromApi = ApiRepository.getThreads(mailboxUuid, id).data?.threads
+            ?.map { threadFromApi ->
+                threadFromApi.initLocalValues()
+                // TODO: Put this back (and make it work) when we have EmbeddedObjects
+                // threadsFromRealm.find { it.uid == threadFromApi.uid }?.let { threadFromRealm ->
+                //     threadFromApi.messages.forEach { messageFromApi ->
+                //         threadFromRealm.messages.find { it.uid == messageFromApi.uid }?.let { messageFromRealm ->
+                //             messageFromApi.apply {
+                //                 fullyDownloaded = messageFromRealm.fullyDownloaded
+                //                 body = messageFromRealm.body
+                //                 attachments = messageFromRealm.attachments
+                //             }
+                //         }
+                //     }
+                // }
+                // threadFromApi
+            }
+            ?: emptyList()
+
+        // Get outdated data
+        Log.d("API", "Threads: Get outdated data")
+        val deletableThreads = threadsFromRealm.filter { fromRealm ->
+            !threadsFromApi.any { fromApi -> fromApi.uid == fromRealm.uid }
+        }
+        val deletableMessages = deletableThreads.flatMap { thread -> thread.messages.filter { it.folderId == id } }
+
+        // Save new data
+        Log.i("API", "Threads: Save new data")
+        threads = threadsFromApi.toRealmList()
+        MailboxContentController.upsertFolder(this)
+
+        // Delete outdated data
+        Log.e("API", "Threads: Delete outdated data")
+        deletableMessages.forEach { MailboxContentController.deleteMessage(it.uid) }
+        deletableThreads.forEach { MailboxContentController.deleteThread(it.uid) }
+    }
+
+    fun getRole(): FolderRole? = when (_role) {
```

Comment:
You might as well make FolderRole.valueOf

Reply:
And no need for a get method either since we are in kotlin, especially since the role attribute is free